From 9f0ec9ed2e5239b84e2757642fcfa064db0852c2 Mon Sep 17 00:00:00 2001
From: Liu Linhong <linhong.liu@intel.com>
Date: Thu, 25 Jun 2015 13:32:30 +0800
Subject: [PATCH] Report link is inactive when this happen

Inactive means the USB3 link is broken, report this can help debug.
Also this is requested for SSIC failure statistic in crashtool.

Change-Id: Ieb312dc3ca60dccf0cc8e2354178de27e027f6f3
Signed-off-by: Liu Linhong <linhong.liu@intel.com>
---
 drivers/usb/host/xhci-ring.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/host/xhci-ring.c b/drivers/usb/host/xhci-ring.c
index 48c42372bf1b..1db8fd13f3da 100644
--- a/drivers/usb/host/xhci-ring.c
+++ b/drivers/usb/host/xhci-ring.c
@@ -1740,8 +1740,10 @@ static void handle_port_status(struct xhci_hcd *xhci,
 		usb_hcd_resume_root_hub(hcd);
 	}
 
-	if (hcd->speed == HCD_USB3 && (temp & PORT_PLS_MASK) == XDEV_INACTIVE)
+	if (hcd->speed == HCD_USB3 && (temp & PORT_PLS_MASK) == XDEV_INACTIVE) {
+		xhci_warn(xhci, "SS port %d goes into inactive\n", port_id);
 		bus_state->port_remote_wakeup &= ~(1 << faked_port_index);
+	}
 
 	if ((temp & PORT_PLC) && (temp & PORT_PLS_MASK) == XDEV_RESUME) {
 		xhci_dbg(xhci, "port resume event for port %d\n", port_id);
-- 
1.9.1

